src/platform/net/zephyr/http.c

```diff
@@ -370,10 +385,20 @@ http_response_cb(struct http_response *response, MENDER_ARG_UNUSED enum http_fin
             mender_log_error("An error occurred, stop reading data");
         }
     }
+#if ZEPHYR_VERSION_BELOW_FOUR_TWO
+    return;
+#else
+    return 0;
+#endif /* ZEPHYR_VERSION_BELOW_FOUR_TWO */
```

Comment:
We could do
```
#if ZEPHYR_VERSION_BELOW_FOUR_TWO
#define HTTP_CALLBACK_RETURN_VALUE
#else
#define HTTP_CALLBACK_RETURN_VALUE 0
#endif
```
and use the latter macro without and `#if-#else`. But not a big difference...

Reply:
The same applies to `HTTP_CALLBACK_RETURN_TYPE`. And while I think about the two macros, I prefer them over having `#if-#else` in multiple places. Keep that contained in one place where we check the Zephyr version macros and define our own based on it.